src/utils/validations/component/validations/with-validation.hoc.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { ValidationsContext } from './form-with-validations.hoc';
+import Icon from '../../../../components/icon';
+import validator from '../../validator';
+
+const withValidation = (WrappedComponent) => {
+  class WithValidation extends React.Component {
+    static contextType = ValidationsContext;
+
+    static propTypes = {
+      children: PropTypes.node, // Children elements
+      name: PropTypes.string, // Name to uniquely identify the component
+      value: PropTypes.string, // The current value of the component
+      error: PropTypes.oneOfType([ // The error validations that should be run against the value
+        PropTypes.func,
+        PropTypes.arrayOf(PropTypes.func)
+      ]),
+      validationTypes: PropTypes.arrayOf(PropTypes.string), // the types of validations to be run on the component
+      warning: PropTypes.oneOfType([ // The warnings validations that should be run against the value
```

Comment:
this should be `warnings` for backwards compat

Reply:
Ok will change it back